libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/ImageSettingsDialogFragment.java

```diff
@@ -353,22 +362,65 @@ private JSONObject extractMetaDataFromFields(JSONObject metaData) {
 
         return metaData;
     }
+
     /**
      * Loads the given network image URL into the {@link NetworkImageView}.
      */
-    private void loadThumbnail(String imageUrl, NetworkImageView imageView) {
-        if (imageUrl != null) {
-            Uri uri = Uri.parse(imageUrl);
-            String filepath = uri.getLastPathSegment();
+    private void loadThumbnail(String imageUrl) {
+        if (imageUrl == null) {
+            AppLog.e(AppLog.T.MEDIA, "Image url is null! Show the default error image.");
+            showErrorImage();
+        }
 
-            if (MediaUtils.isValidImage(filepath)) {
-                imageView.setImageUrl(imageUrl, mImageLoader);
-            }
+        Uri uri = Uri.parse(imageUrl);
+        String filepath = uri.getLastPathSegment();
+
+        if (MediaUtils.isValidImage(filepath)) {
+            int size = DisplayUtils.dpToPx(
+                    getActivity(),
+                    getResources().getDimensionPixelSize(R.dimen.image_settings_dialog_thumbnail_size)
+            );
+            mImageView.setVisibility(View.GONE);
+            mProgress.setVisibility(View.VISIBLE);
+
+            mImageLoader.get(imageUrl, new ImageLoader.ImageListener() {
+                @Override
+                public void onResponse(ImageLoader.ImageContainer response, boolean isImmediate) {
+                    if (!isAdded())  {
+                        return;
+                    }
+                    if (response.getBitmap() != null) {
+                        mImageView.setVisibility(View.VISIBLE);
+                        mProgress.setVisibility(View.GONE);
```

Comment:
We want to hide the progress bar even when the bitmap is null. We should also hide it in `onErrorResponse` below.